Sources/StreamChatUI/Gallery/GalleryVC.swift

```diff
@@ -400,12 +400,28 @@ open class GalleryVC:
     /// - Returns: An item to share.
     open func shareItem(at indexPath: IndexPath) -> Any? {
         guard let item = getItem(at: indexPath) else { return nil }
-        
-        if let image = item.attachment(payloadType: ImageAttachmentPayload.self) {
-            return image.imageURL
-        } else if let video = item.attachment(payloadType: VideoAttachmentPayload.self) {
-            return video.videoURL
-        } else {
+
+        switch item.type {
+        case .image:
+            let cell = attachmentsCollectionView
+                .cellForItem(at: indexPath) as? ImageAttachmentGalleryCell
+            return cell?.imageView.image
+        case .video:
+            guard let itemAttachment = item.attachment(payloadType: VideoAttachmentPayload.self),
+                  let urlData = try? Data(contentsOf: itemAttachment.videoURL) else {
+                return nil
+            }
+
+            let fileName = itemAttachment.id.messageId.lowercased()
+            let filePath = NSTemporaryDirectory().appending("\(fileName).mov")
```

Comment:
@nuno-vieira Why are we locking to `.mov` format only? Don't we get mime type/extension along with the remote resource?

Android leverages eg `.mp4` format as seen eg [here](https://github.com/GetStream/stream-chat-android/blob/f44d0a5e5b2c324ea7ff8868875e09dda7c3cc0e/stream-chat-android/src/main/java/com/getstream/sdk/chat/utils/CaptureController.java#L35).

Reply:
Makes sense 👍 I will parse the file extension type nice catch!